tests/integration-tests/tests/update/test_update.py

```diff
@@ -529,3 +529,96 @@ def _test_max_vcpus(region, stack_name, vcpus):
 def _test_min_vcpus(region, stack_name, vcpus):
     asg_min_size = get_batch_ce_min_size(stack_name, region)
     assert_that(asg_min_size).is_equal_to(vcpus)
+
+
+@pytest.mark.dimensions("us-west-1", "c5.xlarge", "centos7", "sge")
+@pytest.mark.usefixtures("os", "instance")
+def test_sit_update_compute_instance(region, scheduler, pcluster_config_reader, clusters_factory, s3_bucket_factory):
+    # check case disable_hyperthreading = true, update compute instance
+    _check_update_compute(
+        pcluster_config_reader,
+        clusters_factory,
+        "pcluster.config.disable_ht.ini",
+        "pcluster.config.disable_ht.update.ini",
+        scheduler,
+        region,
+        disable_hyperthreading=True,
+    )
+
+    # check case extra_json is {"cfn_scheduler_slots" : "cores"}, update compute instance
+    _check_update_compute(
+        pcluster_config_reader,
+        clusters_factory,
+        "pcluster.config.extra_json.cores.ini",
+        "pcluster.config.extra_json.cores.update.ini",
+        scheduler,
+        region,
+        cfn_scheduler_slots="cores",
+    )
+
+
+def _check_update_compute(
+    pcluster_config_reader,
+    clusters_factory,
+    init_config_name,
+    update_config_name,
+    scheduler,
+    region,
+    disable_hyperthreading=False,
+    cfn_scheduler_slots=None,
+):
+    # Create cluster with initial configuration
+    init_config_file = pcluster_config_reader(config_file=init_config_name)
+    cluster = clusters_factory(init_config_file)
+
+    # Check cfn_scheduler_slots before changing compute instance type
+    _check_compute_node_slots(
+        cluster,
+        scheduler,
+        region,
+        "c5.xlarge",
+        disable_hyperthreading=disable_hyperthreading,
+        cfn_scheduler_slots=cfn_scheduler_slots,
+        nodes_number=2,
+    )
+    cluster.stop()
+
+    # Update cluster with new configuration and change the compute instance type to c5.2xlarge
+    updated_config_file = pcluster_config_reader(update_config_name)
+    cluster.config_file = str(updated_config_file)
+    cluster.update()
+
+    # Check cfn_scheduler_slots changed by changing compute instance type
+    _check_compute_node_slots(
```

Comment:
I'm wondering how this is working if you don't start the cluster.

Reply:
After update, the scheduler will use autoscaling to initialize the compute nodes, we needn't start the cluster. I tested it and passed. Correct me if I am wrong.